source/Calamari/ArgoCD/Git/RepositoryFactory.cs

```diff
@@ -46,30 +50,53 @@ RepositoryWrapper CheckoutGitRepository(IGitConnection gitConnection, string che
             if (gitConnection.Username != null && gitConnection.Password != null)
             {
                 options.FetchOptions.CredentialsProvider = (url, usernameFromUrl, types) => new UsernamePasswordCredentials
-                    {
-                        Username = gitConnection.Username!,
-                        Password = gitConnection.Password!
+                {
+                    Username = gitConnection.Username!,
+                    Password = gitConnection.Password!
                 };
             }
 
-            var repoPath = Repository.Clone(gitConnection.Url, checkoutPath, options);
+            string repoPath;
+            log.InfoFormat("Cloning repository {0}", log.FormatLink(gitConnection.Url));
+            using (var timedOp = log.BeginTimedOperation("Cloning repository"))
+            {
+                try
+                {
+                    repoPath = Repository.Clone(gitConnection.Url, checkoutPath, options);
+                    timedOp.Complete();
+                }
+                catch (Exception e)
+                {
+                    log.ErrorFormat("Failed to clone Git repository at {0}. Are you sure this is a Git repository?", gitConnection.Url);
```

Comment:
I THINK if we wrap this exception message (or just its message) as a CommandException then that causes the deployment to fail with a clean error message and no stack trace.